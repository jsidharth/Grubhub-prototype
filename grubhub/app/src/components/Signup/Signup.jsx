import React, { Component } from "react";
import {connect} from "react-redux";
import {userActions} from '../../js/actions/index';

class Signup extends Component {
  constructor() {
    super();
    this.state = {
      first_name: '',
      last_name: '',
      email: '',
      password: '',
      type: ''
    };
  }
  
  handleChange = (e) => {
    this.setState({
      [e.target.id]: e.target.value
    });
  }
  handleSignUp = (e) => {
    e.preventDefault();
    const payload = this.state;
    this.props.addUser(payload);
  }
  render() {
    return (
      <div className="container shadow p-3 singup_form">
        <form onSubmit={(e) => this.handleSignUp(e)}>
        <div className="form-group">
            <label htmlFor="firstName">First Name</label>
            <input
              type="text"
              className="form-control"
              id="first_name"
              placeholder="First Name"
              onChange = {this.handleChange}
            />
          </div>
          <div className="form-group">
            <label htmlFor="lastName">Last Name</label>
            <input
              type="text"
              className="form-control"
              id="last_name"
              placeholder="Last Name"
              onChange = {this.handleChange}
            />
          </div>
          <div className="form-group">
            <label htmlFor="email">Email address</label>
            <input
              type="email"
              className="form-control"
              id="email"
              aria-describedby="emailHelp"
              placeholder="Enter email"
              onChange = {this.handleChange}></input>
            <small id="emailHelp" className="form-text text-muted">
              We'll never share your email with anyone else.
            </small>
          </div>
          <div className="form-group">
            <label htmlFor="password">Password</label>
            <input
              type="password"
              className="form-control"
              id="password"
              placeholder="Password"
              onChange = {this.handleChange}
            />
          </div>
          <label>User type:</label>
          <div className="form-check">
            <input
              className="form-check-input"
              type="radio"
              name="userTypeRadio"
              id="type"
              value="Customer"
              defaultChecked
              onChange = {this.handleChange}
            />
            <label className="form-check-label" htmlFor="customerRadio">
              Customer
            </label>
          </div>
          <div className="form-check">
            <input
              className="form-check-input"
              type="radio"
              name="userTypeRadio"
              id="type"
              value="Owner"
              onChange = {this.handleChange}
            />
            <label className="form-check-label" htmlFor="ownerRadio">
              Owner
            </label>
          </div>
          <button type="submit" className="btn btn-primary m-3">
            Sign Up
          </button>
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch, ownProps) => ({
  addUser: payload => dispatch(userActions.addUser(payload, ownProps))
});

export default connect(null, mapDispatchToProps)(Signup);
