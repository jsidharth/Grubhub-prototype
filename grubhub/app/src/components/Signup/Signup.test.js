// Link.react.test.js
import React from 'react';
import {Signup} from './Signup';

import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Signup component', function(){
    it('should render correctly', () => {
      const component = shallow( <Signup/>);
      expect(component).toMatchSnapshot();
    });
    it('should be wrapped in a container div', () => {
        const component = shallow( <Signup/>);
        expect(component.find('.singup_form').length).toEqual(1);
      });
    
      it('should have one Sign up button', () => {
        const component = shallow( <Signup/>);
        expect(component.find('.sign-up-btn').length).toEqual(1);
      });
    
      it('should have first name field', () => {
        const component = shallow( <Signup/>);
        expect(component.find('.first_name').length).toEqual(1);
      });

})

