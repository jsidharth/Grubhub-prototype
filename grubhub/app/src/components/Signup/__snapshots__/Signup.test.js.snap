// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component should render correctly 1`] = `
<div
  className="container shadow p-3 singup_form"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="firstName"
      >
        First Name
      </label>
      <input
        className="form-control first_name"
        id="first_name"
        onChange={[Function]}
        pattern="[a-z A-z]+"
        placeholder="First Name"
        required={true}
        title="Only Alphabets"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="lastName"
      >
        Last Name
      </label>
      <input
        className="form-control"
        id="last_name"
        onChange={[Function]}
        pattern="[a-z A-z]+"
        placeholder="Last Name"
        required={true}
        title="Only Alphabets"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email address
      </label>
      <input
        aria-describedby="emailHelp"
        className="form-control"
        id="email"
        onChange={[Function]}
        placeholder="Enter email"
        required={true}
        type="email"
      />
      <small
        className="form-text text-muted"
        id="emailHelp"
      >
        We'll never share your email with anyone else.
      </small>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
    </div>
    <label>
      User type:
    </label>
    <div
      className="form-check"
    >
      <input
        className="form-check-input"
        defaultChecked={true}
        id="type"
        name="userTypeRadio"
        onChange={[Function]}
        type="radio"
        value="Customer"
      />
      <label
        className="form-check-label"
        htmlFor="customerRadio"
      >
        Customer
      </label>
    </div>
    <div
      className="form-check"
    >
      <input
        className="form-check-input"
        id="type"
        name="userTypeRadio"
        onChange={[Function]}
        type="radio"
        value="Owner"
      />
      <label
        className="form-check-label"
        htmlFor="ownerRadio"
      >
        Owner
      </label>
    </div>
    <button
      className="btn btn-primary m-3 sign-up-btn"
      type="submit"
    >
      Sign Up
    </button>
  </form>
</div>
`;
